firstvoices/backend/views/dictionary_views.py

```diff
@@ -52,3 +57,116 @@ def get_queryset(self):
             )
         else:
             return DictionaryEntry.objects.none()
+
+
+@extend_schema_view(
+    list=extend_schema(
+        description="Returns a word of the day for the given site.",
+        responses={
+            200: DictionaryEntryDetailSerializer,
+            403: OpenApiResponse(description="Todo: Not authorized for this Site"),
+            404: OpenApiResponse(description="Todo: Site not found"),
+        },
+    ),
+)
+class WordOfTheDayView(
+    FVPermissionViewSetMixin,
+    SiteContentViewSetMixin,
+    mixins.ListModelMixin,
+    viewsets.GenericViewSet,
+):
+    """
+    Picks and returns word-of-the-day for a given site.
+    """
+
+    http_method_names = ["get"]
+    serializer_class = DictionaryEntryDetailSerializer
+    pagination_class = None
+
+    @staticmethod
+    def get_suitable_queryset(site_slug):
+        # Case 1. Check if there is a word assigned word-of-the-day date of today
+        today = datetime.today()
+        selected_word = WordOfTheDay.objects.filter(
+            site__slug=site_slug, date=today
+        ).first()
+        if selected_word is not None:
+            queryset = DictionaryEntry.objects.filter(
+                id=selected_word.dictionary_entry.id
+            )
+            if queryset.count() > 0:
+                return queryset
+
+        # Case 2. If no words found with today's date, Get words which have not yet been assigned word-of-the-day
+        words_used = WordOfTheDay.objects.filter(site__slug=site_slug).values_list(
+            "dictionary_entry_id", flat=True
+        )
+        queryset = DictionaryEntry.objects.filter(
+            site__slug=site_slug,
+            type=DictionaryEntry.TypeOfDictionaryEntry.WORD,
+            exclude_from_wotd=False,
+        ).exclude(id__in=list(words_used))
+        if queryset.count() > 0:
+            selected_word = queryset.first()
+            WordOfTheDay(
+                date=today, dictionary_entry=selected_word, site=selected_word.site
+            ).save()
+            return queryset
```

Comment:
Could this queryset potentially return more than one word as the word of the day?

Reply:
Yes, I went with the function definition of returning a queryset here. The first word is selected and then sent back in the response in the `list` function below.
But that may change after discussion with Sarah about refactoring the get_queryset function.